### PR TITLE
iox-#743  iox-#743 Update quality declaration

### PR DESCRIPTION
**Pull request by MatthiasKillat**
Signed-off-by: Matthias Killat <matthias.killat@apex.ai>

## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes **TBD**

### CODE REVIEW

File: QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -23,17 +23,17 @@ On Git, the tags have a `v` prefix before the version numbers. A [release script
 
 `iceoryx` is at a stable version, i.e. `>= 1.0.0`. The latest valid release can be found on the [release page](https://github.com/eclipse-iceoryx/iceoryx/releases) of iceoryx.
 
-The change history can be found in the [release notes section](https://iceoryx.io/latest/release-notes).
+The change history can be found in the [release notes section](doc/website/release-notes/iceoryx-v2-0-0.md).
```
Comment (MatthiasKillat) [inline QUALITY_DECLARATION.md:26]: The existing link was broken but ideally we want to link latest here. Can we do this and if so how? @dkroenke

**Review by elfenpiff — DISMISSED** [review]

File: QUALITY_DECLARATION.md (hunk 1 above)
Comment (dkroenke) [reply QUALITY_DECLARATION.md:26]: I will rework this accordingly, please leave the link here as-is

Comment (MatthiasKillat) [reply QUALITY_DECLARATION.md:26]: @dkroenke you mean I should undo the change I presume?

File: QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -29,11 +29,11 @@ The change history can be found in the [release notes section](https://iceoryx.i
 
 All symbols in the installed headers are considered part of the public API.
 
-Except for the exclusions listed below, all installed headers are in the include directory of the package, headers in any other folders are not installed and considered private. Headers under the folder `internal` are not considered part of the public API and are subject to change without notice.
+Except for the exclusions listed below, all installed headers are in the include directory of the package, headers in any other folders are not installed and considered private. Headers in the folder `internal` are not considered part of the public API and are subject to change without notice.
 
 ### API Stability Within a Released ROS Distribution [1.iv]/[1.vi]
 
-`iceoryx` guarantees public API stability for MINOR and PATCH releases and can not guarantee public API stability for MAJOR releases.
+`iceoryx` guarantees public API stability for MINOR and PATCH releases and cannot guarantee public API stability for MAJOR releases.
```
Comment (FerdinandSpitzschnueffler) [inline QUALITY_DECLARATION.md:36]: I think we usually just write iceoryx and not `iceoryx`. Can you please correct this here and in the whole file?

File: QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -2,18 +2,18 @@ This document is a declaration of the software quality for the `iceoryx_hoofs`,
...
-See [the Readme](https://github.com/eclipse-iceoryx/iceoryx#quality-levels--platforms) for a full list of the quality levels of iceoryx components and [Contributing Guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#quality-levels) for a description of the quality levels.
+See the [Readme](https://github.com/eclipse-iceoryx/iceoryx#quality-levels--platforms) for a full list of the quality levels of iceoryx components and [Contributing Guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#quality-levels) for a description of the quality levels.
 
 ## Version Policy [1]
 
 ### Version Scheme [1.i]
 
 `iceoryx` follows the [semver](https://semver.org/#summary) specification for versioning. The `rmw` implementation [rmw_iceoryx](https://github.com/ros2/rmw_iceoryx) follows the same rules.
-Changes for MINOR releases are API compatible and ideally ABI stable if not otherwise clearly stated in the release notes
+Changes for MINOR releases are API compatible and ideally ABI stable if not clearly stated otherwise in the release notes
```
Comment (FerdinandSpitzschnueffler) [inline QUALITY_DECLARATION.md:16]:
```suggestion
Changes for MINOR releases are API compatible and ideally ABI stable if not clearly stated otherwise in the release notes.
```

File: QUALITY_DECLARATION.md [hunk 4]
```diff
@@ -120,22 +120,22 @@ A CI job ensures by checking with `ament_copyright` that all files comply with t
...
+Every iceoryx package has a `test` folder that contains subfolders for [module- and/or integrationtests](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#testing) written based on the Google test framework. All tests are running on the CI for every Pull-Request for all supported platforms and are contained in the master branch.
 Currently, due to limited support on Windows, there are some exclusions for testing on the CI.
 
 ### Feature Testing [4.i]
 
-The features of iceoryx are tested by the [iceoryx_integrationtest](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_integrationtest) package. Using `launch_test` are tests on system-level executed to ensure that the iceoryx packages are functional and the Public API usable.
-There is continuous effort ongoing to cover the corner cases in the usage of iceoryx.
+The features of iceoryx are tested in the [iceoryx_integrationtest](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_integrationtest) package. Using `launch_test` tests are executed on system-level to ensure that the iceoryx packages are functional and the Public API works according to specification.
```
Comment (FerdinandSpitzschnueffler) [inline QUALITY_DECLARATION.md:128]:
```suggestion
The features of iceoryx are tested in the [iceoryx_integrationtest](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_integrationtest) package. Using `launch_test`, tests are executed on system-level to ensure that the iceoryx packages are functional and the Public API works according to specification.
```

File: QUALITY_DECLARATION.md [hunk 5]
```diff
@@ -120,22 +120,22 @@ A CI job ensures by checking with `ament_copyright` that all files comply with t
...
+There is continuous effort to cover the corner cases in the usage of iceoryx in test-cases.
 
 ### Public API Testing [4.ii]
 
-For every major feature all are tests executed. New features must provide unit and integration tests that cover the code changes in the Pull-Request. The tests reside in separated folders for every package following a defined structure and naming convention.
-The features are tested on module(unit) -integration and system test level. [Guidelines](https://iceoryx.io/latest/advanced/best-practice-for-testing/) for Contributors ensure high quality for test development.
+All tests are executed for every major feature. New features must provide unit and integration tests that cover the code changes in the Pull-Request. The tests reside in separated folders for every package following a defined structure and naming convention.
+The features are tested in module(unit) -integration and system test level. The [guidelines](https://iceoryx.io/latest/advanced/best-practice-for-testing/) for Contributors ensure a high quality of test development.
```
Comment (FerdinandSpitzschnueffler) [inline QUALITY_DECLARATION.md:134]: Isn't it `at ... test level`?

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]
